src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -29,8 +29,23 @@ public CounterConfiguration(CounterFilter filter)
 
     internal static class TraceEventExtensions
     {
+        private static Dictionary<(string, string), CachedCounterInfo> counterInfoByName = new();
         private static HashSet<string> inactiveSharedSessions = new(StringComparer.OrdinalIgnoreCase);
 
+        // This assumes uniqueness of provider/counter combinations;
+        // this is currently a limitation (see https://github.com/dotnet/runtime/issues/93097 and https://github.com/dotnet/runtime/issues/93767)
+        public static CachedCounterInfo GetOrCreateProvider(string providerName, string counterName, string meterTags = null, string instrumentTags = null, string scopeHash = null)
```

Comment:
`GetCounterInfo` or (if my suggested rename is agreed upon) `GetCounterMetadata`. The method should indicate what it is doing, but it is not getting or creating a provider.